### DISCUSSION TITLE
Comparison to langserve

### DISCUSSION THREAD

**Discussion by pietz** [ajndkr/lanarky#143] [General]
It looks like langserve does something very similar. Any comments on your direction of taking this package? I see you plan support for multiple chain/aget/llm libraries. Anything else where you think lanarky comes in?

**Comment by ajndkr** [comment]
Hi @pietz!

That's a great question! Currently, Lanarky and Langserve do share many similar features. However, it's important to note that Langserve is a relatively new library. I started building Lanarky a few months ago when there weren't many options available for deploying an LLM application with streaming capabilities. Langchain, on the other hand, has had the advantage of having numerous contributors, some of whom are working on it full-time, allowing them to catch up quickly! 😅

Despite this, I still have plans for this project because I don't see it as entirely redundant. One significant limitation I see in Langserve is its vendor lock-in with Langchain. Not every LLM application required an agentic approach and there are scenarios where a more streamlined/lean approach (eg: hitting the OpenAI API directly) is better for production.

With these considerations in mind, my plans for Lanarky are to steer it towards becoming a web framework for LLM applications. This will involve the development of middlewares, routers, and other components that are more likely to be used in various LLM applications aiming for production-level performance.

Hope this helps! 🙌
